h2/src/test/org/h2/test/jdbc/TestConnection.java

```diff
@@ -42,6 +42,8 @@ public void test() throws Exception {
         testSetInternalProperty();
         testSetInternalPropertyToInitialValue();
         testSetGetSchema();
+        testCommitOnAutoCommitSet();
+        testRollbackOnAutoCommitSet();
```

Comment:
This all looks very good, but the tests aren't testing the non-default case right now, I think we need:

        SysProperties.FORCE_AUTOCOMMIT_OFF_ON_COMMIT = true
        try {
            testCommitOnAutoCommitSet();
            testRollbackOnAutoCommitSet();
        } finally {
                    SysProperties.FORCE_AUTOCOMMIT_OFF_ON_COMMIT = false;
        }
        testCommitOnAutoCommitSet();
        testRollbackOnAutoCommitSet();

to test both ways.

Thanks!

Reply:
All SysProperties fields are "final", so I can't test it modifying FORCE_AUTOCOMMIT_OFF_ON_COMMIT value. 
I cannot change the value using reflection because removing final modifier is not possible in JDK-12. 
Aren't there any other cases like this in other tests?

Reply:
This field is not strongly required to be final, but if you want to preserve this modifier, take a look on `org.h2.test.unit.TestMemoryUnmapper` that spawns a child process. Don't forget to check its exit code.

Reply:
You need to pass `-ea` (to enable assertions) and `-Dh2.forceAutoCommitOffOnCommit=true` to the JVM and some argument to a program that can be used in the `main(String[])` method to execute only the required method. In case of exception / error child process will be terminated with non-zero exit code and you can compare result of `Process.waitFor()` with zero to ensure that child process was terminated normally. `TestMemoryUnmapper` uses a more complicated logic with different exit codes that is not needed here.

Reply:
i'll go removing final modifier